drivers/vcnl40x0/vcnl40x0_saul.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_vcnl40x0
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for VCNL40X0 device
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "vcnl40x0.h"
+#include "vcnl40x0_params.h"
+#include "xtimer.h"
+
+static int read_proximity(void *dev, phydat_t *res)
+{
+    vcnl40x0_t *d = (vcnl40x0_t *)dev;
+
+    res->val[0] = (int16_t)vcnl40x0_read_proximity(d);
+    res->unit = UNIT_M;
+    res->scale = 0;
```

Comment:
looking into the [datasheet](http://www.vishay.com/docs/83462/vcnl4010.pdf) page 3 there is a log-log graph showing the relation of `cts` to `mm` for distinct LED currents. It seems one could derive an approximation for distances between 1 to 100 mm, which would need 2 fixed points for calibration, i.e., cts at 1mm and cts at 100mm distance. 

Sadly, I don't have the sensor in close reach to help you on that. So we might live with cts for now and add conversion function later on.